MerlinAU.sh

```diff
@@ -522,18 +522,22 @@ logo() {
   echo -e "${NOct}"
 }
 
-##-----------------------------------------------##
-## Modified by: ExtremeFiretop [2023-Dec-16]     ##
-##-----------------------------------------------##
+##----------------------------------------##
+## Modified by Martinski W. [2024-Jun-05] ##
+##----------------------------------------##
 _CheckForNewScriptUpdates_()
 {
    local DLRepoVersionNum  ScriptVersionNum
 
    echo ""
+   [ -s "$SCRIPTVERPATH" ] && DLRepoVersion="$(cat "$SCRIPTVERPATH")"
+   rm -f "$SCRIPTVERPATH"
+
    # Download the latest version file from the source repository
-   curl --silent --retry 3 "${SCRIPT_URL_BASE}/version.txt" -o "$SCRIPTVERPATH"
+   curl -LSs --retry 4 --retry-delay 5 "${SCRIPT_URL_BASE}/version.txt" -o "$SCRIPTVERPATH"
 
-   if [ ! -f "$SCRIPTVERPATH" ] ; then UpdateNotify=0 ; return 1 ; fi
+   if [ $? -ne 0 ] || [ ! -s "$SCRIPTVERPATH" ]
+   then scriptUpdateNotify=0 ; return 1 ; fi
 
    # Read in its contents for the current version file
    DLRepoVersion="$(cat "$SCRIPTVERPATH")"
```

Comment:
Any thoughts on added a check for this variable?

    # Read the contents of the version file
    DLRepoVersion="$(cat "$SCRIPTVERPATH" 2>/dev/null)"
    if [ -z "$DLRepoVersion" ]; then
        echo "Downloaded version file is empty."
        UpdateNotify=0
        return 1
    fi

Reply:
> Any thoughts on added a check for this variable?
> 
> ```
> # Read the contents of the version file
> DLRepoVersion="$(cat "$SCRIPTVERPATH" 2>/dev/null)"
> if [ -z "$DLRepoVersion" ]; then
>     echo "Downloaded version file is empty."
>     UpdateNotify=0
>     return 1
> fi
> ```
>
I don't have any objections to adding the extra check. However, note that the newly-merged code already checks if the "version.txt" file is empty. So this extra check would be if the `cat` command fails somehow, so the error message is not that the file is empty, but that the variable is empty.

`echo "Variable for downloaded version is empty."`

My 2 cents.